### PR TITLE
Add {merge_commit,squash_commit}_template

### PR DESCRIPTION
**Pull request by borancar**
### PR Checklist Acknowledgement

- [x] I acknowledge that all of the following items are true, where applicable:
  - Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
  - [Examples](https://github.com/gitlabhq/terraform-provider-gitlab/tree/main/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
  - New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
  - No new `//lintignore` comments were copied from existing code. (Linter rules are meant to be enforced on new code.)

### CODE REVIEW

**Comment by timofurrer** [comment]
This is partly addressed in #917 already. However, I may merge this priorly if #917 will take more time and we'll release.

This feature (from whichever PR) will go into the next release for sure.

Thanks 🎉

**Comment by borancar** [comment]
> This is partly addressed in #917 already. However, I may merge this priorly if #917 will take more time and we'll release.
> 
> This feature (from whichever PR) will go into the next release for sure.
> 
> Thanks tada

Sounds good to me, I'm just adding tests now to complete the PR.

**Comment by borancar** [comment]
Acceptance tests pass locally for me, could a maintainer approve workflows? Just so we can leave this ready in case it ends up the PR that's merged.vs #917

**Review by timofurrer — APPROVED** [review]

**Pull request closed without merging**
